### DISCUSSION TITLE
Come evitare di aspettare mezzanotte per ascoltare Alexa...

### DISCUSSION THREAD

**Discussion by Dregi56** [indomus/forum#218] [HUB: Home Assistant]
Seguendo la guida inDomus ho implementato garbage_collection per la verifica del calendario di conferimento della raccolta rifiuti differenziata. Mediante l'automazione suggerita dalla guida, aggiorno una scheda Lovelance, ma io vorrei anche una notifica vocale.
La guida fa osservare: _Ovviamente il cambio di stato avviene ogni giorno alla mezzanotte, orario non proprio adatto per farsi notificare l’eventuale, prossima raccolta differenziata. In caso si preferisca essere allertati ad un altro orario sarà necessario adottare un blocco action diverso, il quale interpreti lo stato dei sensori e notifichi quel che si crede_.

Mi sembra che la soluzione sia utilizzare uno script da lanciare ad un orario stabilito. Non so se una automazione possa essere sfruttata da uno script, mi sembra di capire che invece possa essere fatto il contrario. Quindi dovrei trasformare il codice dell'automazione che verifica i sensori in modo automatico (a mezzanotte) appunto, in uno script che li verifichi ad una chiamata...
mi date qualche suggerimento, non so da che parte iniziare...

**Comment by dragorex71** [comment]
Ma se più semplicemente imposti un delay (ritardo) tra le action della automazione? Se metti 8 ore ti avvertirà alle 8 di mattina per esempio.
Ecco le istruzioni per il comando
https://www.home-assistant.io/docs/scripts/#delay
Sono riferite ad uno script ma vanno bene anche nelle automazioni

**Reply by Dregi56** [reply]
certo, a questo avevo pure pensato, ma sarei comunque legato ad un delay che io vedo metodo valido per un attesa di qualche minuto, al limite qualche ora, ma ad esempio volendo la notifica ad ora di cena sarei costretto ad un delay di 20 ore.... 
Per questo cercavo qualcosa di diverso....
Avevo pensato all'affidamento del "valore" ad una variabile ambiente, verificabile al momento opportuno....

**Reply by essegi65** [reply]
Io non uso alexa, ma google home comunque penso sia la solita cosa. Ho creato un'automazione che scatta all'ora desiderata e come condizione ho messo lo stato del sensore del rito della raccolta a "tomorrow" (difatti lo devo mettere fuori la sera prima) e in caso positivo mi manda l'avviso su google home.

**Reply by Dregi56** [reply]
_lo stato del sensore del rito della raccolta_  puoi tradurre? Io ho quattro sensori: carta, umido, vetro, plastica.
nella mia ideale flow chart metterei:
if  state sensor.carta is tomorrow then
    tipo= carta
if state sensor.plastica is tomorrow then
   tipo= plastica 
if state sensor.vetro is tomorrow then
  tipo= vetro

Chiamata al servizio di notifica:
"Preparare per domani l'asporto" & tipo

Ma non lo so mettere con il linguaggio per home assistant

**Reply by essegi65** [reply]
Il package GARBAGE una volta installato ti crea i sensori per il materiale da differenziare (sensor.carta, sensor.plastica, ecc.) il loro stato, io uso HASSIO e lo puoi vedere su strumenti per sviluppatori, riporta una data (tra quanti giorni deve essere ritirata la raccolta), il giorno prima lo stato riporterà "tomorrow".
Beni se crei un'automazione con attivazione all'orario che vuoi e metti come condizione che lo stato del sensore sia "tomorrow" farai partire il servizio che attiva alexa a seconda della condizione. Io ho scelto di creare 4 automazioni separate.
Ci sono altri utenti più esperti di me che potranno anche indicarti un unica automazione con più condizioni (sempre su lo stato dei sensori che riportano TOMORROW) e con più action che partono a seconda del tipo di condizione.

**Comment by Dregi56** [comment]
Sto cercando di approcciare uno script del tipo che segue. naturalmente è un abbozzo con ogni probabilità carico di inesattezze e con mancanze di vario tipo, ma è per rendere l'idea di che percorso intenderei fare. Se c'è qualche buon'anima che me lo corregge, magari imparo qualcosa :)
```
waste:
    alias: Differenziata
    sequence:
    {% if is_state('sensor.biologico_umido.days', 'tomorrow') %}
    tipo: biologico
    {% endif %}
    {% if is_state('sensor.secco_indifferenziato.days', 'tomorrow') %}
    tipo: secco indifferenziato
    {% endif %}
    {% if is_state('sensor.carta.days', 'tomorrow') %}
    tipo: carta
    {% endif %}
    {% if is_state('sensor.plastica_vetro.days', 'tomorrow') %}
    tipo: vetro e plastica
    {% endif %}
    service: notify.alexa_media_echoshow_mio
    data:
      message: "Raccolta differenziata: domani viene asportata la frazione "{% tipo %}
      title: Raccolta differenziata
      data:
        data:
          type: announce
          method: spoken
          volume_level: 60
        target: media_player.alexa
  mode: single
```
